registry/client/transport/transport.go

```diff
@@ -0,0 +1,147 @@
+package transport
```

Comment:
@dmcgowan let me know if you disagree with anything there. I suggest we unexport OnEOFReader once we got rid of the AuthTransport in docker.